op-service/dial/active_rollup_provider.go

```diff
@@ -29,6 +29,8 @@ type ActiveL2RollupProvider struct {
 	currentRollupClient RollupClientInterface
 	rollupIndex         int
 	clientLock          *sync.Mutex
+
+	ActiveProviderChanged *sync.Cond
```

Comment:
It's discouraged to use `sync.Cond`, it has a complex usage pattern. Why not just a `changedHook func()` that can be set at construction? That's more versatile, in particular could also contain actions on a `sync.Cond` or a channel or any other synchronization method. In our case, we could just send a signal on a `chan struct{}` that the throttling loop listens on.

Reply:
Yeah I agree. This was actually just something I explored and didn't mean to commit.

Reply:
[2988863](https://github.com/ethereum-optimism/optimism/pull/14244/commits/2988863da57b37f4d3cfbef36acbf11037767a0e)